src/navigation/StackNavigators/CameraStackNavigator.js

```diff
@@ -0,0 +1,216 @@
+
+const androidReadPermissions = usesAndroid13Permissions
+  ? PermissionsAndroid.PERMISSIONS.READ_MEDIA_IMAGES
+  : PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE;
+
+const Stack = createNativeStackNavigator( );
+
+const CameraContainerWithPermission = ( ) => {
+  if ( usesAndroid10Permissions ) {
+    // WRITE_EXTERNAL_STORAGE is deprecated after Android 10
+    // https://developer.android.com/training/data-storage/shared/media#access-other-apps-files
+    return (
+      <PermissionGate
+        permission={PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE}
+      >
+        <PermissionGate permission={PermissionsAndroid.PERMISSIONS.CAMERA}>
+          <CameraContainer />
+        </PermissionGate>
+      </PermissionGate>
+    );
+  }
+  return (
+    <PermissionGate permission={PermissionsAndroid.PERMISSIONS.CAMERA}>
+      <CameraContainer />
+    </PermissionGate>
+  );
+};
+
+const SoundRecorderWithPermission = ( ) => {
+  if ( usesAndroid10Permissions ) {
+    return (
+      <PermissionGate permission={PermissionsAndroid.PERMISSIONS.RECORD_AUDIO}>
+        <PermissionGate
+          permission={PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE}
+        >
+          <PermissionGate
+            permission={androidReadPermissions}
+          >
+            <SoundRecorder />
+          </PermissionGate>
+        </PermissionGate>
+      </PermissionGate>
+    );
+  }
+  return (
+    <PermissionGate permission={PermissionsAndroid.PERMISSIONS.RECORD_AUDIO}>
+      <PermissionGate
+        permission={androidReadPermissions}
+      >
+        <SoundRecorder />
+      </PermissionGate>
+    </PermissionGate>
+  );
+};
+
+const PhotoGalleryWithPermission = ( ) => {
+  if ( usesAndroid10Permissions ) {
+    return (
+      <PermissionGate
+        permission={androidReadPermissions}
+      >
+        <PermissionGate
+          permission={PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE}
+        >
+          <PermissionGate
+            permission={PermissionsAndroid.PERMISSIONS.ACCESS_MEDIA_LOCATION}
+          >
+            <PhotoGallery />
+          </PermissionGate>
+        </PermissionGate>
+      </PermissionGate>
+    );
+  }
+
+  return (
+    <PermissionGate permission={androidReadPermissions}>
+      <PermissionGate
+        permission={PermissionsAndroid.PERMISSIONS.ACCESS_MEDIA_LOCATION}
+      >
+        <PermissionGate permission={PERMISSIONS.IOS.PHOTO_LIBRARY} isIOS>
+          <PermissionGate
+            permission={PERMISSIONS.IOS.LOCATION_WHEN_IN_USE}
+            isIOS
+          >
+            <PhotoGallery />
+          </PermissionGate>
+        </PermissionGate>
+      </PermissionGate>
+    </PermissionGate>
+  );
+};
+
+const ObsEditWithPermission = ( ) => (
+  <Mortal>
+    <PermissionGate
+      permission={PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION}
+    >
+      <ObsEdit />
+    </PermissionGate>
+  </Mortal>
+);
+
+const CameraStackNavigator = ( ): Node => (
```

Comment:
Can we call this the `AddObsStackNavigator` instead? It seems to encompass more than just camera-related screens, but it doesn't seem to handle more than the add obs flow. Alternatively, you could think of each add obs option as an entry point to its own stack, since you can't really navigate between them (e.g. switch to the ARCamera while you're importing photos). That might be excessive, though.